src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```
@@ -504,6 +504,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                                     Value="$(DebuggerSupport)"
                                     Trim="true" />
 
+    <RuntimeHostConfigurationOption Include="System.Diagnostics.Metrics.Meter.IsSupported"
+                                    Condition="'$(MetricsSupport)' != ''"
+                                    Value="$(MetricsSupport)" />
```

```suggestion
                                    Value="$(MetricsSupport)"
                                    Trim="true" />
```

You'll want to pass `Trim=true` here, so it gets flown into the trimmer and AOT compiler.